Content/Pickups/AbilityPickup.cs

```diff
@@ -95,77 +60,57 @@ public sealed override void AI()
 				if (!Fancy)
 					return;
 
-				/*if (Vector2.Distance(Main.screenPosition + new Vector2(Main.screenWidth / 2, Main.screenHeight / 2), NPC.Center) <= Main.screenWidth / 2 + 100) //shader
-                {
-                    float timer = Math.Abs((float)Math.Sin(StarlightWorld.rottime));
-                    Filters.Scene.Activate("Shockwave", NPC.Center).GetShader()
-                        .UseProgress(Main.screenWidth / (float)Main.screenHeight)
-                        .UseIntensity(500 + 200 * (timer))
-                        .UseDirection(new Vector2(0.005f + timer * 0.03f, 1 * 0.008f - timer * 0.004f));
-                }*/
-
-				if (Vector2.Distance(Main.LocalPlayer.Center, NPC.Center) < 200f) //music handling
+				if (Vector2.Distance(Main.LocalPlayer.Center, Center) < 200f) //music handling
 				{
 					for (int k = 0; k < Main.musicFade.Length; k++)
 					{
 						if (k == Main.curMusic)
-							Main.musicFade[k] = Vector2.Distance(Main.LocalPlayer.Center, NPC.Center) / 200f;
+							Main.musicFade[k] = Vector2.Distance(Main.LocalPlayer.Center, Center) / 200f;
 					}
 				}
 			}
 
 			Main.blockInput = false;
 
-			if (mp.pickupTarget?.whoAmI == NPC.whoAmI)
+			if (mp.pickupTarget == this)
 			{
 				PickupVisuals(mp.pickupTimer); //if the Player is picking this up, clientside only also
 				Main.blockInput = true;
 				// TODO sync it so they're not floating? idk
 			}
 		}
 
-		public sealed override bool CanHitPlayer(Player target, ref int cooldownSlot)
+		public override void Collision(Player player)
```

Comment:
make sure you test this in multiplayer. vanilla NPCs only check CanHitPlayer for the current client's player. our dummies run collision for all players on all clients + server. it probably works but it is a functional change to be aware of